src/visual/style.ts

```diff
@@ -179,21 +180,24 @@ const dataColorPaletteTask: StageHandler = {
         // Each type of series use one scope.
         // Pie and funnel are using diferrent scopes
         const paletteScopeGroupByType = createHashMap<object>();
-        ecModel.eachSeries(function (seriesModel) {
-            if (!seriesModel.useColorPaletteOnData) {
+        ecModel.eachSeries((seriesModel: SeriesModel<SeriesOption & ColorByMixin>) => {
+            const colorBy = seriesModel.getColorBy();
+            if (colorBy === 'series') {
                 return;
             }
-            let colorScope = paletteScopeGroupByType.get(seriesModel.type);
+            const key = seriesModel.type + '-' + colorBy;
+            let colorScope = paletteScopeGroupByType.get(key);
             if (!colorScope) {
                 colorScope = {};
-                paletteScopeGroupByType.set(seriesModel.type, colorScope);
+                paletteScopeGroupByType.set(key, colorScope);
             }
             inner(seriesModel).scope = colorScope;
         });
 
 
-        ecModel.eachSeries(function (seriesModel) {
-            if (!seriesModel.useColorPaletteOnData || ecModel.isSeriesFiltered(seriesModel)) {
+        ecModel.eachSeries((seriesModel: SeriesModel<SeriesOption & ColorByMixin>) => {
+            const colorBy = seriesModel.getColorBy();
```

Comment:
The original logic `ecModel.isSeriesFiltered` seems to be removed. I think it's a mistake?